infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/ShellScriptProvider.java

```diff
@@ -61,15 +61,16 @@ public Deployment createInfrastructure(Infrastructure infrastructure, String inf
         String testPlanLocation = Paths.get(infraRepoDir, "DeploymentPatterns" ,
                 infrastructure.getName()).toString();
 
-        logger.info("Creating the Kubernetes cluster...");
+        logger.info("Executing provisioning scripts...");
         try {
-            TestGridUtil.executeCommand("bash " + Paths.get(testPlanLocation,
-                    getScriptToExecute(infrastructure, Script.ScriptType.INFRA_CREATE)), null);
+            ShellExecutor executor = new ShellExecutor(null);
+            executor.executeCommand("bash " + Paths
+                    .get(testPlanLocation, getScriptToExecute(infrastructure, Script.ScriptType.INFRA_CREATE)));
         } catch (CommandExecutionException e) {
             throw new TestGridInfrastructureException("Exception occurred while executing the infra-create script " +
                                                       "for deployment-pattern '" + infrastructure.getName() + "'" , e);
         }
-        return null;
+        return new Deployment();
```

Comment:
Why do you return an empty deployment object here?

Reply:
Deployment object contains the host entries of  the deployment But in the future we are not going to maintain endpoints in TG. So this was done to get the execution flow working.

Reply:
But, when are we going to fix it properly. :-) IMO, it is better to fix the issues then-and-there. If the fix is big, we can raise a new ticket and handle it separately.